services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/models/shared_store.py

```diff
@@ -14,6 +23,40 @@ class SharedStore(BaseModel):
         description="stores the container names from the compose_spec",
     )
 
-    def clear(self):
+    async def clear(self):
         self.compose_spec = None
         self.container_names = []
+        await self.persist_to_disk()
+
+    @classmethod
+    async def init_from_disk(cls, shared_store_dir: Path) -> "SharedStore":
+        data_file_path = shared_store_dir / STORE_FILE_NAME
+        if data_file_path.exists():
+            # if the sidecar is started for a second time (usually the container dies)
+            # it will load the previous data which was stored
+            async with aiofiles.open(shared_store_dir / STORE_FILE_NAME) as data_file:
+                file_content = await data_file.read()
+
+            obj = cls.parse_obj(file_content)
+        else:
+            obj = cls()
+
+        obj._shared_store_dir = shared_store_dir  # pylint: disable=protected-access
+        return obj
+
+    async def persist_to_disk(self) -> None:
```

Comment:
so this guy has now indeed its own state. I think that is good and slowly going towards a more independant dy-sidecar, which will be in the end much simpler and require less interaction with the director-v2. but I'm a bit confused with the naming.

Reply:
Yes, the state is now persisted to the disk.

And yes, the naming is **terrible** and it has been there since the beginning. As said above suggestions are welcomed, I have a mind block in moving away form this name.